msal/application.py

```diff
@@ -220,6 +228,36 @@ def __init__(
             MSAL will combine them into
             `claims parameter <https://openid.net/specs/openid-connect-core-1_0-final.html#ClaimsParameter`_
             which you will later provide via one of the acquire-token request.
+
+        :param str region:
+            Added since MSAL Python 1.12.0.
+
+            As of 2021 May, regional service is only available for
+            ``acquire_token_for_client()`` sent by any of the following scenarios::
+
+            1. An app powered by a capable MSAL
+               (MSAL Python 1.12+ will be provisioned)
+
+            2. An app with managed identity, which is formerly known as MSI.
+               (However MSAL Python does not support managed identity,
+               so this one does not apply.)
+
+            3. An app authenticated by Subject Name/Issuer (SNI).
+
+            4. An app which already onboard to the region's allow-list.
+
+            MSAL's default value is None, which means region behavior remains off.
+            If enabled, some of the MSAL traffic would remain inside that region.
+
+            App developer can opt in to regional endpoint,
+            by provide a region name, such as "westus", "eastus2".
+
+            An app running inside Azure VM can use a special keyword
```

Comment:
>Azure VM  [](http://example.com/codeflow?start=34&length=9)

not just azure vm. Azure PaaS are supported afaik, Neha and Bogdan can keep me honest on the details

Reply:
Yes, it's not just Azure VM. I remember @neha-bhargava tested this on other systems. Azure Functions I think supports this, but may be wrong, @neha-bhargava did a lot of testing here and is a great oracle of information :)

Reply:
Right, auto detection is definitely available on azure functions (Environment variable) and VMs (IMDS call). Since the EAP servers also use the VMs we were able detect the region through IMDS call there. We know cloud services is not supported by imds and env. variable was also not set there. Other than that we don't know yet.